File: frontend/src/services/portfolio.js
import api from "./api";


// ----------------------------
// Buy Stock
// ----------------------------
export async function buyStock(

  email,

  symbol,

  quantity,

  buy_price

) {

  const { data } = await api.post(

    "/portfolio/buy",

    {

      email,

      symbol,

      quantity,

      buy_price

    }

  );

  return data;

}


// ----------------------------
// Sell Stock
// ----------------------------
export async function sellStock(

  email,

  symbol,

  quantity,

  buy_price

) {

  const { data } = await api.post(

    "/portfolio/sell",

    {

      email,

      symbol,

      quantity,

      buy_price

    }

  );

  return data;

}


// ----------------------------
// Portfolio
// ----------------------------
export async function getPortfolio(email) {

  const { data } = await api.get(

    `/portfolio/${email}`

  );

  return data;

}

File: backend/routes/portfolio.py
from fastapi import APIRouter

from models.portfolio import PortfolioItem

from services.portfolio_service import (
    buy_stock,
    sell_stock,
    get_portfolio
)

router = APIRouter(
    prefix="/portfolio",
    tags=["Portfolio"]
)


@router.post("/buy")
def buy(item: PortfolioItem):
    return buy_stock(item)


@router.post("/sell")
def sell(item: PortfolioItem):
    return sell_stock(item)


@router.get("/{email}")
def get(email: str):
    return get_portfolio(email)